datauri/datauri.py

```diff
@@ -55,16 +55,22 @@ def parse(uri):
     """
     if not uri.startswith('data:'):
         raise DataURIError('invalid data uri')
+
     s = uri[5:]
     if not s or ',' not in s:
         raise DataURIError('invalid data uri')
+
     media_type, _, raw_data = s.partition(',')
-    is_base64_encoded = media_type.endswith(';base64')
-    if is_base64_encoded:
+    if media_type.endswith(';base64'):
         media_type = media_type[:-7]
+
+        # https://github.com/eclecticiq/python-data-uri/issues/7
+        raw_data = raw_data.strip()
```

Comment:
this will only deal with whitespace at the beginning and end. it would not handle the ‘whitespace in the middle’ case, which is what #7 is (at least partly) about as well.

Reply:
In #7 the example *does* have whitespace at the beginning of the data. `raw_data` here is the part after `base64,`.

Reply:
https://github.com/eclecticiq/python-data-uri/issues/7#issuecomment-409881851:

> However, in the context of HTML 4 and HTML 5, linefeeds within an element attribute value (such as the "src" above) are ignored[citation needed]. So the data URI above would be processed ignoring the linefeeds, giving the correct result. But note that this is an HTML feature, not a data URI feature, and in other contexts, it is not possible to rely on whitespace within the URI being ignored.

Reply:
Just to be clear - are you proposing that we remove *all* whitespace inside a base64 encoded string to facilitate this, including whitespace in the middle? I'd be okay with that and can implement that, though I think just stripping the start/end of the string facilitates most use cases.

Reply:
Well then we're back to where we started - do you suggest a change to my PR or not? :D

As written in my original comment in #7, I feel like stripping leading/trailing is the safest thing to do. Do you disagree and think we should strip *all* whitespace, or do you disagree and think we shouldn't strip *any* whitespace?

Reply:
heh.  i personally think code should either be strict (no whitespace, per spec) or loose (strip all whitespace, useful for html-ish data), not something in between.

Reply:
(let's move to #7 for discussion)